front/src/modules/timesStops/TimesStopsInput.tsx

```diff
@@ -85,12 +88,34 @@ const TimesStopsinput = ({ allWaypoints, startTime, pathSteps }: TimesStopsInput
       }
       return step;
     });
+
     dispatch(updatePathSteps({ pathSteps: updatedPathSteps }));
   };
+
+  // Handle 00:00:00 time input interpretation as J+1 (next day)
+  const timeWithDayAdjustment = useMemo(
```

Comment:
I think you just need to use the existing `updateDaySinceDeparture`
3fa20ae043ac03b0ab0f911b25a7d2e0aa369113

Reply:
write a few tests like this one 3fa20ae043ac03b0ab0f911b25a7d2e0aa369113

Reply:
done !